README.md

```diff
@@ -48,95 +100,375 @@ Options:
 
-- **link**: Defines [links](https://manual.nexusformat.org/nxdl_desc.html#linktype) between nodes.
+**NeXus link in YAML format**
+```yaml
+reference_measurement(link):
+  target: /NXentry
+  doc: A link to a full data collection.
+```
 
-- **units**: Introduces [NeXus-compliant `NXDL` units](https://manual.nexusformat.org/nxdl-types.html#nxdl-units) arguments, such as `NX_VOLTAGE`.
+In the provided YAML example, `reference_measurement` is defined as a link refering the `NXentry` group with its target specified as `/NXentry`. This structure ensures that the concept referencing the data is effectively linked to the designated target, thereby reducing redundancy and maintaining data integrity within the NeXus framework.
 
-- **dimensions**: Details the expected dimensional arrays.
+### NeXus Choice
+NeXus `choice` concept is designed to choice a concept from a bunch of concepts of the same kind (e.g. NeXus field). The `choice` opens the door to define a scientific concept in several mode regarding different situations.
+
+**NeXus choice in YAML format**
+```yaml
+pixel_shape(choice):
+  (NXoff_geometry):
+    doc: Shape description of each pixel. Use only if all pixels in the detector
+      are of uniform shape.
+  (NXcylindrical_geometry):
+    doc: Shape description of each pixel. Use only if all pixels in the detector
+      are of uniform shape and require being described by cylinders.
+```
 
-- **dim**: A shorthand notation for dimensions, e.g., `(n, )` for a 1D array of length `n` or `(n, m)` for a 2D array of size `n x m`.
+In this choice example `pixes_shape` could be anyone from groups `(NXoff_geometry)` and `(NXcylindrical_geometry)` which are depens on the geometry of pixel.
 
-- **enumeration**: A Python list of strings considered as recommended entries.
+## Special Keywords in YAML
+To explain the context of NeXus, certain keywords hold significance beyond their literal interpretations. These special keywords are utilized to elucidate and denote various NeXus terms e.g. attributes, fields, links, and groups, thereby enhancing the clarity and specificity of the data representation.
+
+### Keyword `exists`
+The `exists` keyword plays a pivotal role in delineating the optionality of NeXus concepts `attribute`, `field`, `choice` `link`, and `group`, during the implementation of NeXus definitions on NeXus files. It provides crucial insights into the expected presence or absence of these concepts within the NeXus data structure. By defalut all the concepts of a base class are optional, and required application definition.
+
+Presently, the accepted values for the `exists` keyword encompass:
+
+`optional`: Denotes that the NeXus concept is not mandatory and may be absent.
+`recommended`: Suggests that the NeXus concept is advisable but not mandatory.
```

Comment:
```suggestion
`recommended`: Suggests that the NeXus concept is advisable, but not mandatory.
```

We were also thinking that recommeded could somehow become a synonym for "FAIR", i.e., any recommended field should also be there if you want to consider your data FAIR. But this is not yet fully fleshed out.